### ISSUE TITLE
Support window.navigator.hardwareConcurrency

### ISSUE THREAD

**Issue by natlibfi-arlehiko** [josdejong/workerpool#6]
Currently, if **maxWorkers** is not defined, the maximum number of workers is *the number of CPU's minus one* expect for browsers for which the number defaults to 3.

I propose **window.navigator.hardwareConcurrency** is to be used in browsers instead. This property is not yet available in all browsers so the previous functionality should be used as a fallback:

```js
var numCPUs = (environment == 'browser') ? window.navigator.hardwareConcurrency || 4 : node.require('os').cpus().length;
this.maxWorkers = Math.max(numCPUs - 1, 1);
```

**Comment by josdejong** [comment]
Thanks! I didn't know about `navigator.hardwareConcurrency`. Nice little improvement.

**Comment by josdejong** [comment]
This should now be available in `v1.3.0`, enjoy.
